src/VisualStudio/CSharp/Test/Options/AutomationObjectTests.cs

```diff
@@ -0,0 +1,279 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.ColorSchemes;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.LanguageServices.CSharp.Options;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Options;
+using Microsoft.VisualStudio.LanguageServices.UnitTests;
+using Microsoft.VisualStudio.LanguageServices.UnitTests.CodeModel;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.CSharp.UnitTests.Options
+{
+    [UseExportProvider]
+    public class AutomationObjectTests
+    {
+        [Fact]
+        public void TestEnsureProperStorageLocation()
+        {
+            using var workspace = TestWorkspace.CreateCSharp("");
+            workspace.GlobalOptions.OptionChanged += (sender, e) =>
+            {
+                if (e.Option.Name is "RefactoringVerification" or "RenameTracking")
+                {
+                    return;
+                }
+
+                Assert.True(e.Option.StorageLocations.Any(s => s is RoamingProfileStorageLocation or LocalUserProfileStorageLocation or FeatureFlagStorageLocation), $"Option '{e.Option.Name}' doesn't have proper storage location for persistence.");
+            };
+
+            var automationObject = new AutomationObject(workspace);
+            var automationObjectType = typeof(AutomationObject);
+            foreach (var property in automationObjectType.GetProperties())
+            {
+                Assert.True(property.CanRead, $"'{property.Name}' must have a getter.");
+                Assert.True(property.CanWrite, $"'{property.Name}' must have a setter.");
+
+                if (property.Name is "WarnOnBuildErrors" or
```

Comment:
Extract to a string array field.

Reply:
@mavasani I plan to remove these options completely in this PR.

The comment here:

https://github.com/dotnet/roslyn/blob/8fd033cd948a2eb33e4af650b7a1b3dd18e914d2/src/VisualStudio/CSharp/Impl/Options/AutomationObject/AutomationObject.ObsoleteAndUnused.cs#L12-L18

says the option is kept for backcompat, but based on what I tried, I don't see any backcompat issues. Probably it was an issue in VS at some point?

To elaborate more, I exported my current settings, modified the output xml to include an option name that doesn't exist (some completely random name), then imported the modified xml again.

If there were backcompat issue, I guess VS would have shown some error that it cannot load this file or something like that. But VS seems to ignore the option it doesn't know about rather than showing an error.

If someone can confirm there are no backcompat concerns, I can remove it in this PR. Otherwise, I'll take the suggestion and extract it to a field.